tag.go

```diff
@@ -69,6 +69,11 @@ func getLatestAcceptableTag(tagConstraint string, tags []string) (string, *Fetch
 		}
 	}
 
+    // check contstraint against only the latest acceptable version
+    if !constraints.Check(latestAcceptableVersion) {
+        return latestTag, wrapError(errors.New("Tag does not exist"))
+    }
+
```

Comment:
@brikis98 here is the test output. Out of habit I was using spaces for indentation not tabs :/. Unfortunately my editor doesn't show a difference till I push to github. I think I'll close this pull request and re-submit after squashing all the commits. How does that sound?
 
--- FAIL: TestDownloadGitTagZipFile (0.46s)
        file_test.go:46: Failed to download file: 500 - Failed to download file at the url https://api.github.com/repos/gruntwork-io/fetch-test-private/zipball/v0.0.2. Received HTTP Response 404.
--- FAIL: TestDownloadGitBranchZipFile (0.26s)
        file_test.go:80: Failed to download file: 500 - Failed to download file at the url https://api.github.com/repos/gruntwork-io/fetch-test-private/zipball/sample-branch. Received HTTP Response 404.
--- FAIL: TestDownloadGitCommitFile (0.56s)
        file_test.go:145: Failed to download file: 500 - Failed to download file at the url https://api.github.com/repos/gruntwork-io/fetch-test-private/zipball/676cfb92b54d33538c756c7a9479bfc3f6b44de2. Received HTTP Response 404.
--- FAIL: TestExtractFiles (0.00s)
        file_test.go:241: Failed to extract files: open test-fixtures/fetch-test-public-0.0.1.zip: no such file or directory
--- FAIL: TestGetListOfReleasesFromGitHubRepo (0.18s)
        github_test.go:29: error fetching releases: 404 - Received HTTP Response 404 while fetching releases for GitHub URL https://github.com/gruntwork-io/fetch-test-private. Full HTTP response: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
--- FAIL: TestGetGitHubReleaseInfo (0.09s)
        github_test.go:155: Failed to fetch GitHub release info for repo 0f5c10a8809aa61251302000f9079183f62cfe4e due to error: 404 - Received HTTP Response 404 while fetching releases for GitHub URL https://github.com/gruntwork-io/fetch-test-private. Full HTTP response: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
--- FAIL: TestDownloadReleaseAsset (0.07s)
        github_test.go:191: Failed to download asset %!s(int=1872521) to /tmp/test-download-release-asset546055932 from GitHub URL https://github.com/gruntwork-io/fetch-test-private due to error: 404 - Received HTTP Response 404 while fetching releases for GitHub URL https://github.com/gruntwork-io/fetch-test-private. Full HTTP response: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
FAIL
exit status 1